apps/server/src/services/RestoreService.ts

```diff
@@ -0,0 +1,124 @@
+
+                const maybePlayback = elements[0] as Playback
+                if (!Object.values(Playback).includes(maybePlayback)) { this.hasValidData = false; return; }
+                this.playback = maybePlayback;
+
+                const maybeId = elements[1] === 'null' ? null : elements[1];
+                //TODO: Cannot access 'EventLoader' before initialization
+                // if (maybeId && EventLoader.getEventWithId(maybeId)) { this.hasValidData = false; return; }
+                this.selectedEventId = maybeId;
+
+                const maybeStartedAt = elements[2] === 'null' ? null : parseInt(elements[2]);
+                if (maybeStartedAt !== null && Number.isNaN(maybeStartedAt)) { this.hasValidData = false; return; }
+                this.startedAt = maybeStartedAt;
+
+                const maybeAddedTime = elements[3] === 'null' ? null : parseInt(elements[3]);
+                if (maybeAddedTime !== null && Number.isNaN(maybeAddedTime)) { this.hasValidData = false; return; }
+                this.addedTime = maybeAddedTime;
+
+                const maybePausedAt = elements[4] === 'null' ? null : parseInt(elements[4]);
+                if (maybePausedAt !== null && Number.isNaN(maybePausedAt)) { this.hasValidData = false; return; }
+                this.pausedAt = maybePausedAt;
+
+                this.hasValidData = true;
+            } else {
+                this.hasValidData = false;
+            }
+        } catch (err) {
+            logger.info(LogOrigin.Server, 'faild to open restore.csv, ' + err);
+            this.hasValidData = false;
+        }
+        this.file = new Writer(path.join(this.filePath, 'restore.csv'));
+    }
+
+    /**
+    * @param {Playback} playback
+    * @param {string} selectedEventId
+    * @param {number} startedAt
+    * @param {number} addedTime
+    * @param {number} pausedAt
+    */
+    async save(playback: Playback, selectedEventId: string | null, startedAt: number | null, addedTime: number | null, pausedAt: number | null) {
+        const newStore = `${playback},${selectedEventId},${startedAt},${addedTime},${pausedAt},\n`;
+        if (newStore != this.lastStore) {
+            this.lastStore = newStore;
+            this.file.write(newStore).catch((err) => { logger.error(LogOrigin.Server, err) });
+        }
+    }
+
+    /**
+    * try to restore timer state from csv
+    */
+    restore() {
+        if (!this.hasValidData) return;
+        switch (this.playback) {
+            case (Playback.Armed):
+                PlaybackService.loadById(this.selectedEventId);
+                break;
+            case (Playback.Pause):
+            case (Playback.Play):
+                if (PlaybackService.loadById(this.selectedEventId)) {
+                    const event = EventLoader.getEventWithId(this.selectedEventId);
+                    eventTimer.init(event, this.playback, this.selectedEventId, this.startedAt, this.addedTime, this.pausedAt);
+                }
```

Comment:
I dont think it is in the scope of this function to do this. This is likely to cause issues down the line (for example with the current code the `eventLoader `is out of sync)

How about creating a `resume()` function in the Playback service that does this?
the function itself should be roughly based in the existing `startById`, with the difference that it passes the remaining data to event timer

In that case, the function in event timer might as well be called resume, since that is more descriptive of what it achieves